app/models/async_transaction/base.rb

```diff
@@ -25,6 +26,14 @@ def parsed_metadata
     validates :transaction_id,
               uniqueness: { scope: :source, message: 'Transaction ID must be unique within a source.' }
 
+    def delete_stale
+      Base.where(
+        'created_at < ?', DELETE_COMPLETED_AFTER.ago
+      ).where(
+        status: Base::COMPLETED
+      ).find_each(&:destroy!)
```

Comment:
Since the `!` version of `destroy` is being called here, perhaps add some error handling to deal with the exception that is raised, where ever this `delete_stale` method is being called (i.e. in the `delete_old_transactions_job.rb`).

Reply:
This was an adapted from an [existing worker](https://github.com/department-of-veterans-affairs/vets-api/blob/d3c610b71c41c624885a5802394aa684a4eadece/app/workers/delete_attachment_job.rb#L10-L18) that does almost the exact same thing. I was trying to conform to that pattern, but after thinking about it more, I don't think it actually makes sense for this to use the `destroy!` in the first place. So I removed the `!`. I think this resolves the naming convention concerns, but let me know if you still think extra logic is required.